mysql-ballerina/build.gradle

```diff
@@ -278,22 +278,75 @@ publishing {
     }
 }
 
+def checkExecResult(execResult, failText, standardOutput){
+    if (execResult) {
+        if (execResult.getExitValue() != 0) {
+            throw new GradleException('Non-zero exit value: ' + execResult.getExitValue())
+        }
+        if (standardOutput.toString().contains(failText)){
+            throw new GradleException('"' + failText + '" string in output: ' + standardOutput.toString())
+        }
+    } else {
+        throw new GradleException('Returned a null execResult object')
+    }
+}
+
+task createTestDockerImage(type: Exec) {
+    if (!Os.isFamily(Os.FAMILY_WINDOWS)) {
+        def standardOutput = new ByteArrayOutputStream()
+        commandLine 'sh', '-c', "docker build -f $project.projectDir/tests/resources/Dockerfile -t ballerina-mysql" +
+                " -q $project.projectDir/tests/resources/"
+        doLast {
+            checkExecResult(execResult, 'Error', standardOutput)
+            sleep(10 * 1000)
+        }
+    }
+}
+
+
+task startTestDockerContainer(type: Exec) {
+     if (!Os.isFamily(Os.FAMILY_WINDOWS)) {
+         def standardOutput = new ByteArrayOutputStream()
+         commandLine 'sh', '-c', "docker run --rm -d --name ballerina-mysql -p 3305:3306 -t ballerina-mysql"
+         doLast {
+             checkExecResult(execResult, 'Error', standardOutput)
```

Comment:
Add another check to ping if the container is up and running else we will get intermittent test failures if the container is not fully initialised. Few times I have noticed this takes more than 20 seconds.

Reply:
> Add another check to ping if the container is up and running else we will get intermittent test failures if the container is not fully initialised. Few times I have noticed this takes more than 20 seconds.

AFAIK no need to ping to check the container because checkExecResult function here exists only after the container is created. 
https://docs.gradle.org/current/javadoc/org/gradle/process/ExecResult.html

Reply:
But since we run the docker command in detached mode I think it may cause intermittent test failures. will check on that.

Reply:
Ping is added